codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -260,26 +246,19 @@ class ProtocolTestGenerator(
             rust("/* test case disabled for this protocol (not yet supported) */")
             return
         }
-        val fnName = if (expectedShape.hasTrait<ErrorTrait>())
-            "serialize_${operationShape.id.name.toSnakeCase()}_error"
-        else
-            "serialize_${operationShape.id.name.toSnakeCase()}_response"
         writeInline("let expected_output =")
         instantiator.render(this, expectedShape, testCase.params)
         write(";")
-        write(
-            """let http_response = #T(&expected_output).expect("failed to serialize response");""",
-            operationMod.member(fnName)
-        )
         rustTemplate(
             """
-            use #{parse_http_response};
-            let parser = #{op}::new();
-            let parsed = parser.parse_loaded(&http_response);
+            use #{ParseStrictResponse};
+            use #{SerializeHttpResponse};
+            let op = #{op}::new();
+            let http_response = op.serialize(&expected_output).expect("unable to serialize response body");
+            let parsed = op.parse(&http_response);
```

Comment:
ah! I get it now. We'll need to add response-specific code to `aws-smithy-protocol-test` 👍🏻